stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1300,6 +1324,26 @@
       (ok { stacker: stacker,
             unlock-burn-height: new-unlock-ht }))))
 
+;; Add an authorization for a signer key.
+;; When an authorization is added, the `signer-sig` argument is not required
+;; in the functions that use it as an argument.
+;; The `allowed` flag can be used to either enable or disable the authorization.
+;; Only the Stacks principal associated with `signer-key` can call this function.
+;; *New in Stacks 3.0*
+(define-public (set-signer-key-authorization (pox-addr { version: (buff 1), hashbytes: (buff 32)})
```

Comment:
I'm not entirely sure but is there *any* situation in which a signer might revoke authorization? IE - do we need an "allowed" parameter or should this always set the 'signer-key-authorizations' to true?

Is there any unexpected behavior that'd happen as a result of a signer setting then revoking the signature?

Reply:
> I'm not entirely sure but is there any situation in which a signer might revoke authorization? IE - do we need an "allowed" parameter or should this always set the 'signer-key-authorizations' to true?

One use-case is that the signer can create and store a revocation transaction (akin to a key revocation certificate in PGP) that they can just have on-hand in case they believe their signing key (which might live somewhere hard to access) is compromised or rendered unusable.

> Is there any unexpected behavior that'd happen as a result of a signer setting then revoking the signature?

You mean setting and revoking the authorization?  A transaction that expects the authorization would subsequently fail.  If we're worried about signers abusing this by setting and revoking authorizations in rapid succession (e.g. within the same block), we can instead map the authorization to the Stacks block height at which it takes effect, and require that block to be in the future.